components/scream/src/share/tests/kokkos_utils_tests.cpp

```diff
@@ -143,4 +143,55 @@ TEST_CASE("team_utils", "[kokkos_utils]") {
 #endif
 }
 
+TEST_CASE("team_utils_cuda", "[kokkos_utils]")
+{
+#ifdef KOKKOS_ENABLE_CUDA
+  using namespace scream::util;
+  using namespace scream;
+
+  using Device = DefaultDevice;
+  using ExeSpace = typename KokkosTypes<Device>::ExeSpace;
+  using MemberType = typename KokkosTypes<Device>::MemberType;
+
+  const int nk = 128;
+  TeamUtils<ExeSpace> tu_temp(ExeSpaceUtils<ExeSpace>::get_default_team_policy(1, nk));
+  const int num_conc = tu_temp.get_num_concurrent_teams();
+
+  const int saturation_multiplier=10;
```

Comment:
Should test <, exactly =, and > physical concurrency, not just > case. There are conditionals in the impl that depend on these cases.